modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +610,272 @@ private void deleteApplication(String applicationId) throws Exception {
 
         oAuth2RestClient.deleteApplication(applicationId);
     }
+
+    @Test(dependsOnMethods = "testDeleteOrganization")
+    public void createOrganizationsForPaginationTests() throws JSONException {
+
+        List<Map<String, String>> organizations = createOrganizations(NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS);
+
+        if (organizations.size() != NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS) {
+            throw new RuntimeException("Failed to create the expected number of organizations for testing pagination.");
+        }
+    }
+
+    @DataProvider(name = "paginationLimitsDataProvider")
+    public Object[][] paginationLimitsDataProvider() {
+
+        return new Object[][]{
+                {10},
+                {20},
+                {25},
+        };
+    }
+
+    @Test(dataProvider = "paginationLimitsDataProvider", dependsOnMethods = "createOrganizationsForPaginationTests")
+    public void testGetPaginatedOrganizationsWithLimit(int limit) {
+
+        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + QUESTION_MARK + LIMIT_QUERY_PARAM + limit;
+        Response response = getResponseOfGetWithOAuth2(endpointURL, m2mToken);
+
+        validateHttpStatusCode(response, HttpStatus.SC_OK);
+
+        int actualOrganizationCount = response.jsonPath().getList(ORGANIZATIONS_PATH_PARAM).size();
+        int expectedOrganizationCount = Math.min(limit, NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS);
+        Assert.assertEquals(actualOrganizationCount, expectedOrganizationCount);
+
+        String nextLink = response.jsonPath().getString(
+                String.format("links.find { it.%s == '%s' }.%s", REL, LINK_REL_NEXT, HREF));
+
+        String afterValue = null;
+
+        if (nextLink != null && nextLink.contains(AFTER_QUERY_PARAM)) {
+            afterValue = nextLink.substring(nextLink.indexOf(AFTER_QUERY_PARAM) + 6);
+        }
+
+        String storedAfterValue = afterValue;
+
+        if (NUM_OF_ORGANIZATIONS_FOR_PAGINATION_TESTS > limit) {
+            Assert.assertNotNull(storedAfterValue);
+        } else {
+            Assert.assertNull(storedAfterValue);
+        }
+    }
+
+    @DataProvider(name = "paginationLimits")
+    public Object[][] paginationLimits() {
```

Comment:
Can we improve these names of data providers? It is a bit abstract and unclear when another person reads the code.

Reply:
Check other places as well.

Reply:
Done